### PR TITLE
Add support for selecting kvm and systemd labels

### PR DESCRIPTION
**Pull request by rhatdan**
In order to better support kata containers and systemd containers
container-selinux has added new types. Podman should execute the
container with an SELinux process label to match the container type.

Traditional Container process : container_t
KVM Container Process:          containre_kvm_t
PID 1 Init process:             container_init_t

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by giuseppe** [comment]
code LGTM

**Comment by vrothberg** [comment]
Gating is now yet happy:
```
[+0012s] libpod/container_internal.go:1934:17: undefined: selinux.KVMContainerLabels
[+0012s] libpod/container_internal.go:1944:18: undefined: selinux.InitContainerLabels
```

File: libpod/container_internal.go [hunk 1]
```diff
@@ -1910,3 +1925,36 @@ func (c *Container) hasNamespace(namespace spec.LinuxNamespaceType) bool {
...
+func getInitLabel(processLabel string) (string, error) {
+	if processLabel == "" {
+		return "", nil
+	}
+	initLabel, _ := selinux.InitContainerLabels()
+	selinux.ReleaseLabel(initLabel)
+	return swapTypes(initLabel, processLabel)
+}
+
+func swapTypes(source, dest string) (string, error) {
```
Comment (vrothberg) [inline libpod/container_internal.go:1949]: Can we rename it to `swapSELinuxTypes`? `libpod` is a really big package and `swapTypes` sounds very generic.

File: libpod/runtime.go [hunk 2]
```diff
@@ -364,13 +364,17 @@ func makeRuntime(ctx context.Context, runtime *Runtime) (err error) {
...
 	supportsNoCgroups := make(map[string]bool)
+	supportsKVM := make(map[string]bool)
 	for _, r := range runtime.config.Engine.RuntimeSupportsJSON {
 		supportsJSON[r] = true
 	}
 	for _, r := range runtime.config.Engine.RuntimeSupportsNoCgroups {
 		supportsNoCgroups[r] = true
 	}
-
+	/*	for _, r := range runtime.config.Engine.RuntimeSupportsKVM {
```
Comment (vrothberg) [inline libpod/runtime.go:374]: What's going on here?

File: libpod/oci.go [hunk 3]
```diff
@@ -102,6 +102,8 @@ type OCIRuntime interface {
 	// SupportsNoCgroups is whether the runtime supports running containers
 	// without cgroups.
 	SupportsNoCgroups() bool
+	// SupportsKVM indicates container will use  KVM separation
```
Comment (haircommander) [inline libpod/oci.go:105]: nit: extra space between use and KVM

File: libpod/runtime.go [hunk 4]
```diff
@@ -379,7 +383,7 @@ func makeRuntime(ctx context.Context, runtime *Runtime) (err error) {
 		json := supportsJSON[name]
 		nocgroups := supportsNoCgroups[name]
 
-		ociRuntime, err := newConmonOCIRuntime(name, paths, runtime.conmonPath, runtime.config, json, nocgroups)
+		ociRuntime, err := newConmonOCIRuntime(name, paths, runtime.conmonPath, runtime.config, json, nocgroups, supportsKVM[name])
```
Comment (haircommander) [inline libpod/runtime.go:386]: I think we should save the value of this like we do nocgroups and json above

File: libpod/container_internal.go (hunk 1 above)
Comment (rhatdan) [reply libpod/container_internal.go:1978]: Done

File: libpod/oci.go (hunk 3 above)
Comment (rhatdan) [reply libpod/oci.go:106]: Fixed

File: libpod/runtime.go (hunk 4 above)
Comment (rhatdan) [reply libpod/runtime.go]: Removed and figured it out inside of the function.

File: libpod/runtime.go (hunk 2 above)
Comment (rhatdan) [reply libpod/runtime.go]: Oops had not implemented, it is now in containers/common.

**Comment by rhatdan** [comment]
@haircommander @vrothberg @umohnani8 PTAL

**Comment by mheon** [comment]
I feel like there's a risk that we overwrite process labels manually set with `--security-opt` here, but otherwise LGTM

**Comment by mheon** [comment]
Actually, additional concern - if we're not setting the process label right on initial container creation, we'll report the wrong label in `podman inspect`

**Comment by rhatdan** [comment]
This is the point where the labels are being created.

**Comment by rhatdan** [comment]
@mheon I think the security-opt  question, is valid.  I will look to fix.

**Comment by rhatdan** [comment]
@mheon With the latest code I have verified you latest concerns.
```
# ./bin/podman run -ti --runtime=kata -d  fedora echo hello
b5cc716495acba7211ba9aecb4bbfeb164eda5f5dc83c8ac2e4dd09510ccd6dd
sh-5.0# podman inspect -l --format '{{.ProcessLabel}}'
system_u:system_r:container_kvm_t:s0:c302,c910

# ./bin/podman run -ti --security-opt label=type:container_t --runtime=kata -d  fedora echo hello
# podman inspect -l --format '{{.ProcessLabel}}'
system_u:system_r:container_t:s0:c345,c516
```

**Comment by TomSweeneyRedHat** [comment]
LGTM once the unhappy tests are happy

**Comment by rhatdan** [comment]
I just destroyed this PR>  Will need to recreate it tomorrow.

**Comment by rhatdan** [comment]
@baude @mheon @giuseppe @vrothberg @TomSweeneyRedHat @QiWang19 @jwhonce  this one is ready to go in.  PTAL

**Pull request merged**